pkg/controller/azvolumeattachment.go

```diff
@@ -594,9 +634,16 @@ func (r *reconcileAzVolumeAttachment) triggerDetach(ctx context.Context, attachm
 		return err
 	}
 
-	if err := r.detachVolume(ctx, azVolumeAttachment.Spec.UnderlyingVolume, azVolumeAttachment.Spec.NodeName); err != nil {
-		klog.Errorf("failed to detach volume %s from node %s: %v", azVolumeAttachment.Spec.UnderlyingVolume, azVolumeAttachment.Spec.NodeName, err)
-		return err
+	// check if underlying volume attachment exists before proceeding with detachment
+	if azVolumeAttachment.Annotations != nil {
```

Comment:
With this check, the driver will not process the detach if azVolumeAttachment.Annotations == nil. Shouldn't this check be like azVolumeAttachment.Annotations == nil || (azVolumeAttachment.Annotations !=nil && vaExists == "false")?

Reply:
I have changed the annotation logic in whole. It now no longer depends on the value of the annotation in the map but rather its presence.